### ISSUE TITLE
az vm create: creates NSG in wrong location

### ISSUE THREAD

**Issue by irl** [Azure/azure-cli#974]
This works:
```
az resource group create -n wusgroup -l "West US"
az vm create -n test -g wusgroup --image credativ:Debian:8:latest --authentication-type ssh
```
This breaks:
```
az resource group create -n wusgroup -l "East US"
az vm create -n test -g wusgroup --image credativ:Debian:8:latest --authentication-type ssh
```
It creates the NSG in West US and comes back with:
```
At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-debug for usage details. {
  "status": "Failed",
  "error": {
    "code": "ResourceDeploymentFailure",
    "message": "The resource operation completed with terminal provisioning state 'Failed'.",
    "details": [
      {
        "code": "DeploymentFailed",
        "message": "At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-debug for usage details.",
        "details": [
          {
            "code": "BadRequest",
            "message": "{\r\n  \"error\": {\r\n    \"code\": \"InvalidResourceReference\",\r\n    \"message\": \"Resource /subscriptions/00000000-0000-0000-0000-00000000000/resourceGroups/TESTGROUP/providers/Microsoft.Network/networkSecurityGroups/TESTNSG referenced by resource /subscriptions/00000000-0000-0000-0000-00000000000/resourceGroups/testgroup/providers/Microsoft.Network/networkInterfaces/testVMNic was not found. Please make sure that the referenced resource exists, and that both resources are in the same region.\",\r\n    \"details\": [\r\n      {\r\n        \"code\": \"NotFound\",\r\n        \"message\": \"Resource /subscriptions/00000000-0000-0000-0000-00000000000/resourceGroups/TESTGROUP/providers/Microsoft.Network/networkSecurityGroups/TESTNSG not found.\"\r\n      }\r\n    ]\r\n  }\r\n}"
          }
        ]
      }
    ]
  }
}  
```

**Comment by tjprescott** [comment]
This should have been fixed with PR #599. It looks like the template being pulled down is still the old one with the hardcoded value.

**Comment by tjprescott** [comment]
@irl We've fixed the referenced templates and this should be fixed.
